core/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -66,6 +67,12 @@ public static EsThreadPoolExecutor newScaling(String name, int min, int max, lon
         return executor;
     }
 
+    public static EsThreadPoolExecutor newBoundedCached(String name, int poolSize, long keepAliveTime, TimeUnit unit, ThreadFactory threadFactory, ThreadContext contextHolder) {
+        EsThreadPoolExecutor executor = new EsThreadPoolExecutor(name, poolSize, poolSize, keepAliveTime, unit, new LinkedBlockingQueue<>(), threadFactory, new EsAbortPolicy(), contextHolder);
```

Comment:
> why did you decide to change the queue type?

@bleskes The synchronous queue that was used previously will block producers until there is a consumer available; it is basically a handoff queue.

When the generic thread pool was unbounded, this property did not matter because if a producer tried to submit to the queue when all the consumers were occupied, a new consumer would be created and thus the queue backing the generic thread pool was *effectively* non-blocking (a producer would only have to wait as long as it took to spin up a new consumer).

However, when we bound the generic thread pool, this property becomes very important because we do not want to block producers.

A linked blocking queue does not have this property; producers will not block waiting for a consumer.

Thus, the switch.